### PR TITLE
api(wait): introduce RunAndWaitForEvent

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/Playwright/WebSocket.cs [hunk 1]
```diff
@@ -48,78 +48,5 @@ internal WebSocket(IChannelOwner parent, string guid, WebSocketInitializer initi
         public string Url => _initializer.Url;
 
         public bool IsClosed { get; internal set; }
-
-        public async Task<T> WaitForEventAsync<T>(PlaywrightEvent<T> webSocketEvent, Func<Task> action = default, Func<T, bool> predicate = default, float? timeout = default)
```
Comment (dgozman) [inline src/Playwright/WebSocket.cs:52]: What happened to websocket? Do we not produce RunAndWaitForEvent here?

File: src/Playwright.Tests/WorkersTests.cs [hunk 2]
```diff
@@ -125,38 +125,6 @@ public async Task ShouldClearUponNavigation()
             Assert.Empty(Page.Workers);
         }
 
-        [Fact]
-        public async Task WorkerShouldWaitOnClose()
```
Comment (dgozman) [inline src/Playwright.Tests/WorkersTests.cs:129]: What happened to these tests?

File: src/Playwright.Tests/WebSocketTests.cs [hunk 3]
```diff
@@ -184,56 +184,5 @@ public async Task ShouldNotHaveStrayErrorEvents()
             await Page.EvaluateAsync("window.ws.close();");
             Assert.Null(socketError);
         }
-
-        [PlaywrightTest("web-socket.spec.ts", "should reject waitForEvent on socket close")]
```
Comment (dgozman) [inline src/Playwright.Tests/WebSocketTests.cs:188]: What happened to these tests?

**Review by dgozman — APPROVED** [review]

File: src/Playwright/WebSocket.cs (hunk 1 above)
Comment (pavelfeldman) [reply src/Playwright/WebSocket.cs:52]: I nuked it from web socket for now, only listeners there.

File: src/Playwright.Tests/WorkersTests.cs (hunk 2 above)
Comment (pavelfeldman) [reply src/Playwright.Tests/WorkersTests.cs:129]: Same

File: src/Playwright.Tests/WebSocketTests.cs (hunk 3 above)
Comment (pavelfeldman) [reply src/Playwright.Tests/WebSocketTests.cs:188]: Removing feature along with test

**Pull request merged**